PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/referencebook/AzureSdkFeatureDetailPanel.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.azuresdk.referencebook;
+
+import com.intellij.ui.HyperlinkLabel;
+import com.intellij.ui.components.JBLabel;
+import com.intellij.ui.components.JBTabbedPane;
+import com.microsoft.azure.toolkit.intellij.azuresdk.model.AzureSdkArtifactEntity;
+import com.microsoft.azure.toolkit.intellij.azuresdk.model.AzureSdkFeatureEntity;
+import com.microsoft.azure.toolkit.intellij.common.SwingUtils;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import lombok.Getter;
+import org.apache.commons.collections4.CollectionUtils;
+
+import javax.swing.*;
+import java.util.List;
+import java.util.Optional;
+
+public class AzureSdkFeatureDetailPanel {
+    private JBLabel titleLabel;
+    private JPanel descPanel;
+    @Getter
+    private JPanel contentPanel;
+    private JBTabbedPane tabPane;
+    private HyperlinkLabel featureDocLink;
+    private JBLabel descLabel;
+
+    public void setData(final AzureSdkFeatureEntity feature) {
+        AzureTaskManager.getInstance().runLater(() -> {
```

Comment:
shall we here check if current Thread is EDT? and avoid AzureTaskManager.getInstance().runLater?